source/common/router/router.cc

```diff
@@ -453,6 +453,10 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
 
   // Set up stat prefixes, etc.
   request_vcluster_ = route_entry_->virtualCluster(headers);
+  if (request_vcluster_) {
+    std::string vc_name = config_.scope_.symbolTable().toString(request_vcluster_->statName());
```

Comment:
`const std::string &`

I am uncertain if calling toSTring(statName()) is a recommended pattern in worker thread. CC @jmarantz for suggestion

Reply:
@jmarantz @ggreenway Any suggestions here?

Reply:
That will take a symbol table lock. Probably not a good idea.

Does request_vcluster_ contain a string name also? It might, or it could.

Reply:
@jmarantz Thanks for the suggestion. I added a string field to store the VC name.